File: Elmah.Contrib.EntityFramework.Test/EntityErrorLogTest.fs
namespace Elmah.Contrib.EntityFramework.Test

open Elmah.Contrib.EntityFramework

open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type JsonBlobNameResolverTest() =
    [<TestMethod>]
    member this.ConnectionString_Should_Return_Provider_Part() =

        // Arrange
        let providerConectionString = "Data Source=localhost;Initial Catalog=Northwind;Password=Alpine"
        let entityConectionString = sprintf "metadata=res://*/;provider=System.Data.SqlClient;provider connection string=\"%s\";" providerConectionString

        let config = new System.Collections.Generic.Dictionary<string,string>()
        config.["connectionString"] <- entityConectionString
        
        // Act
        let entityErrorLog = new EntityErrorLog(config)

        // Assert
        Assert.AreEqual(entityErrorLog.ConnectionString, providerConectionString)

File: Elmah.Contrib.EntityFramework.Test/EntityErrorLogTest.cs
using System;
using FluentAssertions;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Elmah.Contrib.EntityFramework.Test
{
	[TestClass]
	public class EntityErrorLogTest
	{
		[TestMethod]
		public void ConnectionString_Should_Return_Provider_Part_When_Dictionary_Passed_To_Ctor()
		{
			// Arrange
			const string providerConectionString = "Data Source=localhost;Initial Catalog=Northwind;Password=Alpine";
			string entityConectionString = String.Format("metadata=res://*/;provider=System.Data.SqlClient;provider connection string=\"{0}\";", providerConectionString);

			var config = new System.Collections.Generic.Dictionary<string, string>();
			config["connectionString"] = entityConectionString;

			// Act
			var entityErrorLog = new EntityErrorLog(config);

			// Assert
			entityErrorLog.ConnectionString.Should().Be(providerConectionString);
		}

		[TestMethod]
		public void ConnectionString_Should_Return_Provider_Part_When_String_Passed_To_Ctor()
		{
			// Arrange
			const string providerConectionString = "Data Source=localhost;Initial Catalog=Northwind;Password=Alpine";
			string entityConectionString = String.Format("metadata=res://*/;provider=System.Data.SqlClient;provider connection string=\"{0}\";", providerConectionString);

			// Act
			var entityErrorLog = new EntityErrorLog(entityConectionString);

			// Assert
			entityErrorLog.ConnectionString.Should().Be(providerConectionString);
		}
	}
}